### PR TITLE
Enable functions in views

### PR DESCRIPTION
**Pull request by hchandl**

### CODE REVIEW

File: fdb-sql-layer-core/src/main/java/com/foundationdb/sql/optimizer/ViewCompiler.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (C) 2009-2014 FoundationDB, LLC
```
Comment (nathanlws) [inline fdb-sql-layer-core/src/main/java/com/foundationdb/sql/optimizer/ViewCompiler.java]: 2015

**Comment by nathanlws** [comment]
Looks about to me. Any thoughts @MMcM ?

**Comment by nathanlws** [comment]
Must be good then.

Approve

**Comment by MMcM** [comment]
Sorry for the delay. Does `ViewCompiler ` need to be a `ServerOperatorCompiler`? Is the session important?

**Comment by hchandl** [comment]
Yeah, it looks like that was the problem. This should be working now.

**Comment by MMcM** [comment]
This seems like the appropriate fix for now. Things will have to change when we implement view permissions, because we'll want an `AISBinderContext` with the permissions of the definer when expanding the view, not the invoker. But there's lots of other stuff that goes with that and this is simple enough.

Approve

**Comment by hchandl** [comment]
Okay, this will hopefully be passing everything now. The issue was that some tests don't have a ServerSession to pass through to the ViewCompiler, so they failed with NullPointerExceptions even when the view didn't have functions. The latest commit sidesteps the issue by only constructing the ViewCompiler when the view has functions.

**Comment by MMcM** [comment]
That seems expedient, but we really wouldn't want to limit unit testing to result columns with simple derivations. Perhaps a new card to address that in the future?

**Comment by hchandl** [comment]
Yeah, that makes sense. I did spend some time trying to update tests, but got stuck along the way -- this seemed like a reasonable next best solution, at least for right now. I'll add a card to fix the testing.

**Comment by MMcM** [comment]
Approve

**Pull request merged**